File: AdventOfCode/2019/day2.py
f = open("./day2.txt")
data = f.readlines()[0].split(',')
data = [int(d) for d in data]

data[1] = 12
data[2] = 2

i = 0
while True:
    opcode = data[i]
    if opcode == 1:
        val1 = data[data[i+1]]
        val2 = data[data[i+2]]
        data[data[i+3]]=val1+val2
        i += 4
    elif opcode == 2:
        val1 = data[data[i+1]]
        val2 = data[data[i+2]]
        data[data[i+3]]=val1*val2
        i += 4
    elif opcode == 99:
        break

print(data[0])

File: AdventOfCode/2020/2/day2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main(){
	lines,err := readInput()
	if err!=nil {
		fmt.Println(err)
		return
	}
	result := 0
	for _,line := range(lines) {
		parts := strings.Split(line, ":")
		subs := strings.Split(parts[0]," ")
		nums := strings.Split(subs[0],"-")
		range1,_ := strconv.Atoi(nums[0])
		range2,_ := strconv.Atoi(nums[1])
		count := 0
		for _,c := range(parts[1]) {
			if byte(c)==subs[1][0] {
				count += 1
			}
		}
		if count>=range1 && count<=range2 {
			result += 1
		}
	}
	fmt.Println(result)
}

func readInput() ([]string, error) {
	var lines = make([]string, 0)

	file, err := os.Open("./day2.txt")
	if err!=nil {
		return lines, err
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	scanner.Split(bufio.ScanLines)

	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}
	return lines, nil
}

// ans for part 1 is 383